packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,353 @@
+    private readonly tableResizeObserver: ResizeObserver;
+    private initialColumnPixelWidths: {
+        initialPixelWidth: number,
+        minPixelWidth: number
+    }[] = [];
+
+    private previousTotalTableWidth = 0;
+
+    public constructor(
+        private readonly table: Table<TData>,
+        private readonly virtualizer: Virtualizer<TData>
+    ) {
+        this.virtualizerNotifier = Observable.getNotifier(
+            this.table.virtualizer
+        );
+        this.virtualizerNotifier.subscribe(this, 'headerContainerMarginRight');
+        this.tableResizeObserver = new ResizeObserver(entries => {
+            if (entries[0]?.contentRect.width) {
+                if (
+                    this.previousTotalTableWidth
+                        < entries[0].contentRect.width
+                    && this.table.tableWidthFactor > 1
+                ) {
+                    const pixelDelta = entries[0].contentRect.width
+                        - this.previousTotalTableWidth;
+                    this.table.tableWidthFactor = Math.max(
+                        this.table.tableWidthFactor
+                            - pixelDelta / this.previousTotalTableWidth,
+                        1
+                    );
+                }
+
+                this.previousTotalTableWidth = entries[0].contentRect.width;
+            }
+        });
+        this.headerRowActionContainerResizeObserver = new ResizeObserver(
+            entries => {
+                if (entries[0]) {
+                    this.updateTableViewportMinWidth();
+                }
+            }
+        );
+    }
+
+    public connectedCallback(): void {
+        this.headerRowActionContainerResizeObserver.observe(
+            this.table.headerRowActionContainer!
+        );
+        this.tableResizeObserver.observe(this.table);
+    }
+
+    public getGridTemplateColumns(): string {
+        return this.table.columns
+            ?.filter(column => !column.columnHidden)
+            .map(column => {
+                const {
+                    minPixelWidth,
+                    currentPixelWidth,
+                    currentFractionalWidth
+                } = column.columnInternals;
+                if (currentPixelWidth) {
+                    const coercedPixelWidth = Math.max(
+                        minPixelWidth,
+                        currentPixelWidth
+                    );
+                    return `${coercedPixelWidth}px`;
+                }
+
+                return `minmax(${minPixelWidth}px, ${currentFractionalWidth}fr)`;
+            })
+            .join(' ');
+    }
+
+    /**
+     * Sets up state related to interactively sizing a column.
+     * @param columnIndex The column index currently being hovered over
+     * @param activeColumnDivider The divider that was clicked on (columnIndex - 1 for left divider, columnIndex for right)
+     */
+    public beginColumnInteractiveSize(
+        columnIndex: number,
```

Comment:
Why do you need the `columnIndex` here? I would think knowing which divider was clicked would be enough. For example, what's the difference between clicking on the right divider of column 0 vs the left divider of column 1 (i.e. the difference of passing `0, 0` and `1, 0`)?

Reply:
I've reworked the logic to remove this.